src/main/java/gov/usgs/cida/pubs/transform/CrossrefTransformer.java

```diff
@@ -130,49 +124,72 @@ public void write(Object result) throws IOException {
+				writeModelToTemplate(model, "crossref/body_minimal.ftlx");
+				logMessage = String.format("Created minimal Crossref XML (indexId: %s doi: %s)", pub.getIndexId(), pub.getDoi());
+				LOG.info(logMessage);
+			} catch (TemplateException | IOException e) {
+				logMessage = String.format("Error transforming publication (indexId: %s doi: %s) into minimal Crossref XML.",
+											pub.getIndexId(), pub.getDoi());
+				LOG.error(logMessage, e);
+				writeComment(String.format("Excluded Problematic Publication (indexId: %s doi: %s)", pub.getIndexId(), pub.getDoi()));
+			}
+		}
+	}
 
-				model.put("warehousePage", pubBusService.getWarehousePage(pub));
+	// return the reason the minimal record was created
+	protected String getMinimalRecordComment(Publication<?> pub) {
+		String comment="";
+		boolean isNumberedSeries = PubsUtils.isUsgsNumberedSeries(pub.getPublicationSubtype());
+		if(getCrossrefContributors(pub) == null){
+			comment = "Minimal doi record shown due to publication having no Authors or Editors listed.";
+		} else if(pub.getSeriesTitle() == null || pub.getSeriesTitle().getText() == null) {
+			comment = "Minimal doi record shown due to publication missing series title.";
+		} else if(isNumberedSeries && pub.getSeriesTitle().getOnlineIssn() == null) {
+			comment = "Minimal doi record shown due to publication missing series online issn number.";
+		} else if(isNumberedSeries && pub.getSeriesNumber() == null) {
+			comment = "Minimal doi record shown due to publication missing series number.";
+		} else {
+			comment = "Minimal doi record shown due to error creating full crossref record.";
+		}
+		return comment;
+	}
 
-				model.put("pubContributors", contributors);
+	protected Map<String, Object> makeModel(Publication<?> pub) {
+		Map<String, Object> model = new HashMap<>();
+		model.put("pub", pub);
+		boolean isNumberedSeries = PubsUtils.isUsgsNumberedSeries(pub.getPublicationSubtype());
+		model.put("isNumberedSeries", isNumberedSeries);
 
-				model.put("authorKey", ContributorType.AUTHORS);
-				model.put("editorKey", ContributorType.EDITORS);
-				model.put("compilerKey", ContributorType.COMPILERS);
-				writeModelToTemplate(model, "crossref/body.ftlx");
-				success = true;
-			}
-		} catch (TemplateException | IOException e) {
-			/**
-			 * Since publications are of varying quality, we omit 
-			 * erroneous publications and continue on to the next
-			 * publication.
-			 */
-			String message = getExcludedErrorMessage(pub);
-			LOG.error("Error transforming object into Crossref XML. "
-				+ message, e);
+		model.put("warehousePage", pubBusService.getWarehousePage(pub));
 
-			//add error message as a comment to the xml document
-			writeComment(message);
-		}
-		return success;
+		model.put("pubContributors", getCrossrefContributors(pub));
+
+		model.put("authorKey", ContributorType.AUTHORS);
+		model.put("editorKey", ContributorType.EDITORS);
+		model.put("compilerKey", ContributorType.COMPILERS);
```

Comment:
probably should wait for the other JIRA to add this (and we actually have a lot of compilers in the system)

Reply:
yes, we have PUBSTWO-1721 ticket for add Compiler contributors. This was part of
the previous incomplete implementation of Compiler support. I'll take out compilerKey
of the model.